source/octf/utils/ProtobufReaderWriter.cpp

```diff
@@ -31,47 +35,57 @@ ProtobufReaderWriter::ProtobufReaderWriter(const std::string &filePath)
     } else {
         m_directoryPath = ".";
     }
+
+    openFile();
+}
+
+ProtobufReaderWriter::~ProtobufReaderWriter() {
+    closeFile();
 }
 
 bool ProtobufReaderWriter::read(google::protobuf::Message &message) {
-    // 1. Open and read file
-    std::ifstream file;
-    file.open(m_filePath);
-    if (!file.good()) {
-        // Cannot open file
-        return false;
+    if (m_readFd == -1) {
+        throw Exception("Bad file descriptor for file: " + m_filePath);
     }
 
-    // 2. Read all file content
-    std::stringstream ss;
-    ss << file.rdbuf();
-
-    // 3. Check if read performed well
-    if (!file.good()) {
-        // cannot read file
-        file.close();
-        return false;
+    // Get file size by setting file offset to the end
+    off_t fileSize;
+    fileSize = ::lseek(m_readFd, 0, SEEK_END);
+    if (fileSize == -1) {
+        throw Exception("Could not seek the end of file: " + m_filePath);
+    }
+    // Reset file offset
+    if (::lseek(m_readFd, 0, SEEK_SET) == -1) {
+        throw Exception("Could not seek the start of file: " + m_filePath);
     }
 
-    // File no needed any more, close it
-    file.close();
+    // Create a string to hold file contents
+    std::string fileContent(fileSize, 0);
+
+    // Read the file to string
+    if (::read(m_readFd, &fileContent[0], fileSize) != fileSize) {
+        throw Exception("I/O Error when reading file: " + m_filePath);
+    }
 
-    // 4. Parse JSON string into procol buffer message
+    // Parse JSON string into protocol buffer message
     google::protobuf::util::Status status;
     google::protobuf::util::JsonParseOptions opts;
 
     // We want to have exact fields, when unknown one then we except error
     opts.ignore_unknown_fields = false;
 
-    status = google::protobuf::util::JsonStringToMessage(ss.str(), &message,
+    status = google::protobuf::util::JsonStringToMessage(fileContent, &message,
                                                          opts);
 
     return status.ok();
 }
 
 bool ProtobufReaderWriter::write(const google::protobuf::Message &message) {
-    // 1. First generate string with configuration
+    if (m_writeFd == -1) {
+        return false;
```

Comment:
Please make consistent behavior for read and write paths when file descriptor is not available.